src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -24,18 +24,21 @@ public class ImportSpecification : ISymbolNameSource
         // language=regex
         private const string SemanticVersionPattern = @"(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?";
 
-        private static readonly Regex SpecificationPattern = new(
-            @$"^(?<name>{NamePattern})@(?<version>{SemanticVersionPattern})$",
-            RegexOptions.ECMAScript | RegexOptions.Compiled);
```

Comment:
Shouldn't we keep these?

Reply:
I don't think so, they are a subset of the one we kept. The address resolution is already processed in the semantic processing, so we don't need to duplicate the effort. Rather only we are looking to identify the implicit symbol from the specification string.

Reply:
> The address resolution is already processed in the semantic processing, so we don't need to duplicate the effort.

Could you mention this in a comment in the code - this is good context to share. I was reviewing this with the assumption that "this validation needs to be watertight because we're relying on it".

Regardless, it feels odd to strip out `RegexOptions.ECMAScript | RegexOptions.Compiled` - do you have the context on why they were originally included? (cc @shenglol)

Reply:
`RegexOptions.ECMAScript` prevents Unicode characters from being matched by `\d`, `\w`, etc.

Reply:
Should we remove `SemanticVersionPattern` as well?